packages/sakuli-legacy/src/context/common/test-case/test-case.class.ts

```diff
@@ -1,39 +1,18 @@
 import {cwd} from "process";
-import {Project, TestExecutionContext} from "@sakuli/core";
+import {Project, TestExecutionContext, TestStepContext} from "@sakuli/core";
 import nutConfig from "../nut-global-config.class";
-import {ScreenApi} from "../actions/screen.function";
 import {ifPresent, Maybe, throwIfAbsent} from "@sakuli/commons";
 import {isAbsolute, join} from "path";
 import {TestCase} from "./test-case.interface";
+import {TestStepCache} from "./steps-cache/test-step-cache.class";
+import {takeErrorScreenShot} from "./take-error-screen-shot.function";
 
-type TestMetaData = {
-    suiteName: string,
-    caseName: string
-};
+export function createTestCaseClass(ctx: TestExecutionContext,
+                                    project: Project,
+                                    currentTestFolder: Maybe<string>,
+                                    testStepCache = new TestStepCache()
```

Comment:
The `TestStepCache` uses `os.tmpdir()` by default.
We might consider creating the cache explicitly, using the `currentTestFolder` as storage location.

When backed by a temp folder, the steps cache would get evicted on every reboot (at least), so when running inside a container, the cache would have no effect.